cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -963,7 +976,7 @@ private List<String> getFirewallTargetTags(Network network, boolean useInternalI
       firewalls = compute.firewalls().list(conf.getNetworkHostProjectID()).execute();
     } catch (Exception e) {
       handleRetryableExceptions(e);
-      throw e;
+      throw new ErrorCodeWrappedException(e, ErrorGroup.DATAPROC, ErrorCodeProvider.ERR_NETWORK);
```

Comment:
it would be good for the provisioner to define some base exception class where the group is hardcoded to dataproc so that it doesn't need to be constantly specified.

Reply:
I am not sure if all exceptions thrown with have the same group. For example, sometimes I may want to return CONFIGURATION. I expect we will settle on this once we add the exception in other places. This exception is being thrown in 6 places. Not sure if creating a base exception with not too much other logic in it is worth it still. I would like to not do this for now.

Reply:
DataprocRuntimeException is acting as the base exception now.